File: src/main/java/maple/expectation/global/error/exception/InvalidPotentialGradeException.java
package maple.expectation.global.error.exception;

import maple.expectation.global.error.CommonErrorCode;
import maple.expectation.global.error.exception.base.ClientBaseException;
import maple.expectation.global.error.exception.marker.CircuitBreakerIgnoreMarker;

/**
 * 유효하지 않은 잠재능력 등급 예외
 *
 * <p>CubeCostPolicy에서 잘못된 등급명 입력 시 발생합니다.
 * CircuitBreakerIgnoreMarker를 구현하여 서킷브레이커에 영향을 주지 않습니다.</p>
 *
 * @see maple.expectation.domain.v2.PotentialGrade
 */
public class InvalidPotentialGradeException extends ClientBaseException
        implements CircuitBreakerIgnoreMarker {

    public InvalidPotentialGradeException(String grade) {
        super(CommonErrorCode.INVALID_INPUT_VALUE, "잠재능력 등급: " + grade);
    }
}

File: module-common/src/main/kotlin/maple/expectation/error/exception/InvalidPotentialGradeException.kt
package maple.expectation.error.exception

import maple.expectation.error.CommonErrorCode
import maple.expectation.error.exception.base.ClientBaseException

/**
 * 유효하지 않은 잠재능력 등급 예외 (4xx Client Error)
 *
 * 큐브 사용 시 입력되는 잠재능력 등급이 유효하지 않을 때 발생합니다.
 *
 * @property invalidGrade 유효하지 않은 등급명 (예: "레어", "에픽", "유니크", "레전드리"가 아닌 값)
 */
class InvalidPotentialGradeException(
    invalidGrade: String
) : ClientBaseException(
    CommonErrorCode.INVALID_INPUT_VALUE,
    "잠재능력 등급: $invalidGrade"
)
